src/poetry/repositories/parsers/html_page_parser.py

```diff
@@ -0,0 +1,18 @@
+from __future__ import annotations
+
+from html.parser import HTMLParser
+
+
+class HTMLPageParser(HTMLParser):
+    def __init__(self) -> None:
+        super().__init__()
+        self.base_url: str | None = None
+        self.anchors: list[dict[str, str | None]] = []
+
+    def handle_starttag(self, tag: str, attrs: list[tuple[str, str | None]]) -> None:
+        if tag == "base" and self.base_url is None:
+            base_url = dict(attrs).get("href")
+            if base_url is not None:
+                self.base_url = base_url
```

Comment:
What's the `base_url` about? Is it used anywhere? Can we remove this code block and the attribute?

Reply:
It's for compatibility with HTML specs. I've added the handling for it and the proper tests.